### PR TITLE
scaleway: Fix case mismatch causing key lookup failure

### PR DESCRIPTION
**Pull request by olof**
##### SUMMARY
Since commit f70dc261c, automatic jsonification only happens if the Content-Type header is application/json, but the class only included Content-type in its default set of headers (notice the case variation).

This caused a KeyError on send() if the caller relied on the default content-type value.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
ansible.module_utils.scaleway

##### ADDITIONAL INFORMATION
* I noticed this with the scaleway_compute module, but as it's shared widely among all the scaleway modules, I think more are affected.
* The commit f70dc261c has not been part of any ansible releases. (So, I omitted a changelog fragment, as I think that superfluous? Correct me if I'm wrong.)

### CODE REVIEW

**Review by remyleone — APPROVED** [review]

**Comment by olof** [comment]
I notice that now that my pull request comes in the midst of a [split of the ansible repo](https://github.com/ansible-collections/overview). Are you planning on creating an ansible-collections repo for the scaleway modules? If so, will you rebase this pull request or should I reopen once you've migrated? Same goes for the other pull request I opened. I have some other possible changes on the way, but I'll hold off on those a bit. Thanks! :)

**Comment by remyleone** [comment]
Hi Olof, thanks a lot for your contribution :) At the moment, we are focusing much of our effort on the [CLI](https://github.com/scaleway/scaleway-cli) and the [terraform provider for Scaleway](https://github.com/terraform-providers/terraform-provider-scaleway/). Once those projects are feature-complete we will come back to add support for all new products (Kapsule, LoadBalancer, Baremetal, Database, ...) inside ansible. 

As for the collections, we are interested because it will allow us to have our own schedule for releasing new versions without waiting for ansible maintainers approval. At the moment we haven’t investigated how to migrate our existing modules to ansible collections but we will do so soon.

**Comment by olof** [comment]
So, just to confirm: I'll just wait until you've set that up :).

(The scaleway modules have been removed from this repo, so I will not be able to resolve the `needs_rebase` label.)

**Comment by gundalow** [comment]
@remyleone Hi, the scaleway modules have been migrated to `community.general` https://github.com/ansible-collections/community.general If you've like to move these to their own collection please follow this procedure https://github.com/ansible-collections/overview/blob/master/README.rst#q-what-should-i-do-to-move-plugins-across-collections-during-migration It would be nice to see this plugins moved under https://github.com/scaleway/ as you say, that means you can release at your own schedule. Please do shout out in `#ansible-community` on Freenode IRC if we can help with this

@olof Hi, could you please create a new PR against https://github.com/ansible-collections/community.general/blob/master/plugins/module_utils/scaleway.py then we can get it merged. If a new collection does get created, it will be done by migrating from `community.general`.

**Comment by olof** [comment]
@gundalow thanks, I've opened a new pull request at https://github.com/ansible-collections/community.general/pull/445.

**Pull request closed without merging**
